pyfftw/interfaces/_utils.py

```diff
@@ -63,21 +63,31 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
 
     if calling_func in ('irfft2', 'irfftn'):
         # overwrite_input is not an argument to irfft2 or irfftn
-        args = (a, s, axes, planner_effort, threads, 
-                auto_align_input, auto_contiguous)
+        args = (planner_effort, threads, auto_align_input, auto_contiguous)
 
         if not overwrite_input:
             # Only irfft2 and irfftn have overwriting the input
             # as the default (and so require the input array to 
             # be reloaded).
-            reload_after_transform = True
+            work_with_copy = True
     else:
-        args = (a, s, axes, overwrite_input, planner_effort, threads, 
+        args = (overwrite_input, planner_effort, threads, 
                 auto_align_input, auto_contiguous)
     
+        if not a.flags.writeable:
+            # Special case of a locked array - always work with a
+            # copy.  See issue #92.
+            work_with_copy = True
```

Comment:
We need to be a bit more careful with this. There is a potential conflict between `writeable` and `overwrite_input`. If `writeable` is `False`, `overwrite_input` should forced be `False` (or potentially an exception raised).